### ISSUE TITLE
[Bug]: The engine does dot use the input parameters specified

### ISSUE THREAD

**Issue by sfoucher** [open-edge-platform/anomalib#2633]
### Describe the bug

Hello,

I'm using the latest version on GitHub on Google Colab.

```
from anomalib.metrics import AUROC, F1Score, Evaluator
from anomalib.models import Padim
from torchvision.transforms import v2
from anomalib.pre_processing import PreProcessor
from lightning.pytorch.callbacks import EarlyStopping

pre_processor = Padim.configure_pre_processor(image_size=224)
!rm -rf /content/results
# Create metrics
metrics = [
    AUROC(fields=["pred_score", "gt_label"], prefix="image_"),
    #F1Score(fields=["pred_label", "gt_label"], prefix="image_")
]
print(metrics[0].name)  
evaluator = Evaluator(test_metrics=metrics)

# Create evaluator with metrics
evaluator = Evaluator(val_metrics= metrics, 
                      test_metrics=metrics,
                      # Move computation to CPU for large datasets
                      compute_on_cpu=True)

model = Padim(pre_trained=True,
              evaluator=evaluator,
              pre_processor=pre_processor
              )

print(model.pre_processor.transform)

model.task = TaskType.CLASSIFICATION  # Set the task type here

# Configure model checkpoint callback to monitor a valid metric
checkpoint_callback = ModelCheckpoint(
    dirpath="./checkpoints", 
    save_top_k=1, 
    monitor="image_AUROC", 
    mode="max"
)
early_stopping = EarlyStopping(monitor="image_AUROC", 
                               mode="max", 
                               patience=10)
# Engine setup with training parameters
# https://lightning.ai/docs/pytorch/stable/common/trainer.html#init
engine = Engine(
    max_epochs=20,
    min_epochs=10,
    callbacks=[checkpoint_callback, early_stopping],
    logger=wandb_logger,
    enable_checkpointing=True,  # Save model checkpoints
    log_every_n_steps= 1,
    enable_model_summary= True,
    fast_dev_run= False
)

# Train the model
engine.fit(datamodule=datamodule, model=model)

```

The training starts fine but then stops after one Epochs with the following message:

...
Epoch 0: 100%
 20/20 [00:08<00:00,  2.45it/s, v_num=2f8x]
INFO:lightning_fabric.utilities.rank_zero:`Trainer.fit` stopped: `max_epochs=1` reached.
```

### Dataset

N/A

### Model

PADiM

### Steps to reproduce the behavior

See code above, another dataset could be used

### OS information

OS information:
- OS: Ubuntu 22.04.4 LTS (Google Colab)
- Python version: 3.11
- Anomalib version: 2.1.0.dev0
- PyTorch version: 2.6.0+cu124
- CUDA/cuDNN version: cu124
- GPU models and configuration:1 T4 (16 Gb)
- Any other relevant information:  I'm using a custom dataset


### Expected behavior

According to the engine parameters (max_epochs=20, min_epochs=10), it should run for at least 10 epochs but it is stopping because max_epochs=1

### Screenshots

_No response_

### Pip/GitHub

GitHub

### What version/branch did you use?

2.1.0.dev0

### Configuration YAML

```yaml
Used python code
```

### Logs

```shell
image_AUROC
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/utilities/parsing.py:209: Attribute 'pre_processor' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['pre_processor'])`.
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/utilities/parsing.py:209: Attribute 'evaluator' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['evaluator'])`.
INFO:lightning_fabric.utilities.rank_zero:GPU available: True (cuda), used: True
INFO:lightning_fabric.utilities.rank_zero:TPU available: False, using: 0 TPU cores
INFO:lightning_fabric.utilities.rank_zero:HPU available: False, using: 0 HPUs
INFO:lightning_fabric.utilities.rank_zero:`Trainer(val_check_interval=1.0)` was configured so validation will run at the end of the training epoch..
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/callbacks/model_checkpoint.py:654: Checkpoint directory /content/checkpoints exists and is not empty.
INFO: LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
INFO:lightning.pytorch.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/core/optimizer.py:183: `LightningModule.configure_optimizers` returned `None`, this fit will run with no optimizer
INFO: 
  | Name           | Type          | Params | Mode 
---------------------------------------------------------
0 | pre_processor  | PreProcessor  | 0      | train
1 | post_processor | PostProcessor | 0      | train
2 | evaluator      | Evaluator     | 0      | train
3 | model          | PadimModel    | 2.8 M  | train
---------------------------------------------------------
2.8 M     Trainable params
0         Non-trainable params
2.8 M     Total params
11.131    Total estimated model params size (MB)
16        Modules in train mode
69        Modules in eval mode
INFO:lightning.pytorch.callbacks.model_summary:
  | Name           | Type          | Params | Mode 
---------------------------------------------------------
0 | pre_processor  | PreProcessor  | 0      | train
1 | post_processor | PostProcessor | 0      | train
2 | evaluator      | Evaluator     | 0      | train
3 | model          | PadimModel    | 2.8 M  | train
---------------------------------------------------------
2.8 M     Trainable params
0         Non-trainable params
2.8 M     Total params
11.131    Total estimated model params size (MB)
16        Modules in train mode
69        Modules in eval mode
Compose(
      Resize(size=[224], interpolation=InterpolationMode.BILINEAR, antialias=True)
      Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], inplace=False)
)
Epoch 0: 100%
 20/20 [00:10<00:00,  1.91it/s, v_num=2f8x]
INFO:lightning_fabric.utilities.rank_zero:`Trainer.fit` stopped: `max_epochs=1` reached.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by samet-akcay** [comment]
@sfoucher, this is because Padim model needs only 1 epoch to go through the entire dataset to extract the features. It then performs the anomaly detection on top of the extracted features. Setting `max_epochs=20` would extract the same features 20 times, without impacting the performance. This is why we explicitly set the `max_epochs=1` here in Padim implementation
https://github.com/open-edge-platform/anomalib/blob/7a3bf3aa9ba324b8a76e34606b0cf1e45e283e0a/src/anomalib/models/image/padim/lightning_model.py#L202

if you try another fine-tuning based model, you will be able to set the `max_epochs` and/or other params.

**Comment by sfoucher** [comment]
Ok, thank you for the quick reply, is there another unsupervised anomaly detection model that I could fine-tuned in Padim?

**Comment by samet-akcay** [comment]
There are many such as cfa, cflow, csflow, draem, dsr, efficient_ad, fastflow, fre, reverse_distillation. 
Closing this since it is not an issue. Feel free to create an issue if you come across any other problems. Thanks
